contracts/RockPaperScissors.sol

```diff
@@ -0,0 +1,334 @@
+        game.playerTwo = msg.sender;
+        game.playerTwoMoveHash = moveHash;
+        game.step = Steps.PLAYER_TWO_MOVE;
+        game.expiryDate = expiryDate;
+        depositStake(game.stake);
+
+        emit PlayerTwoMoves(gameKey, msg.sender, msg.value, moveHash, expiryDate);
+    }
+
+    function depositStake(uint256 stake) internal {
+
+        if (msg.value == stake) {
+            return;
+        }
+
+        uint256 senderBalance = balances[msg.sender];
+        if (msg.value < stake) {
+            require(senderBalance >= stake.sub(msg.value), "Insufficient balance");
+        }
+
+        balances[msg.sender] = senderBalance.add(msg.value).sub(stake);
+    }
+
+    function revealPlayerOne(bytes32 secret, uint8 playerOneMove) external whenNotPaused {
+
+        require(Moves(playerOneMove) >= Moves.ROCK && Moves(playerOneMove) <= Moves.SCISSORS, INVALID_MOVE_MSG);
+        bytes32 gameKey = createPlayerOneMoveHash(msg.sender, secret, Moves(playerOneMove));
+        Game storage game = games[gameKey];
+
+        require(game.step == Steps.PLAYER_TWO_MOVE, INVALID_STEP_MSG);
+        game.step = Steps.PLAYER_ONE_REVEAL;
+
+        uint256 expiryDate = block.timestamp.add(FORFEIT_WINDOW);
+        game.expiryDate = expiryDate;
+
+        game.playerOneMove = Moves(playerOneMove);
+
+        emit PlayerOneReveals(gameKey, msg.sender, Moves(playerOneMove), expiryDate);
+    }
+
+    function revealPlayerTwo(bytes32 gameKey, bytes32 secret, Moves playerTwoMove) external whenNotPaused {
+
+        require(Moves(playerTwoMove) > Moves.EMPTY && Moves(playerTwoMove) <= Moves.SCISSORS, INVALID_MOVE_MSG);
+        Game storage game = games[gameKey];
+        require(game.step == Steps.PLAYER_ONE_REVEAL, INVALID_STEP_MSG);
+        address playerTwo = game.playerTwo;
+
+        //Validate move
+        bytes32 expectedMoveHash = createPlayerTwoMoveHash(msg.sender, gameKey, secret, Moves(playerTwoMove));
+        require(game.playerTwoMoveHash == expectedMoveHash, HASH_MISMATCH_MSG);
+
+        emit PlayerTwoReveals(gameKey, msg.sender, Moves(playerTwoMove));
+
+        uint256 stake = game.stake;
+        address playerOne = game.playerOne;
+        address winner;
+        address loser;
+        GameResult outcome = resolveGame(uint8(game.playerOneMove), uint8(playerTwoMove));
+
+        if (outcome == GameResult.DRAW) {
+            resetGame(game);
+            balances[playerOne] = balances[playerOne].add(stake);
+            balances[playerTwo] = balances[playerTwo].add(stake);
+            DrawRefund(gameKey, playerOne, playerTwo, stake, stake);
+
+            return;
+
+        } else if (outcome == GameResult.LEFTWIN) {
+            winner = playerOne;
+            loser = playerTwo;
+        } else if (outcome == GameResult.RIGHTWIN) {
+            winner = playerTwo;
+            loser = playerOne;
+        } else {
+            require(outcome != GameResult.INCORRECT, "Bugger");
```

Comment:
Can you tell me what this `else require(outcome != GameResult.INCORRECT` is doing?

Reply:
😆  Sorry again!! I added that in to help me debug and forgot to remove!

I resolved based on the other conversation.

Reply:
What it was doing was allowing all other wrong values. Only `.INCORRECT` was prevented.